elfpy/markets/hyperdrive/hyperdrive_market.py

```diff
@@ -549,13 +547,13 @@ def add_liquidity(
     def remove_liquidity(
         self,
         agent_wallet: wallet.Wallet,
-        bond_amount: float,
+        lp_shares: float,
     ) -> tuple[hyperdrive_actions.MarketDeltas, wallet.Wallet]:
         """Computes new deltas for bond & share reserves after liquidity is removed"""
         self.apply_checkpoint(self.latest_checkpoint_time, self.market_state.share_price)
         market_deltas, agent_deltas = hyperdrive_actions.calc_remove_liquidity(
             wallet_address=agent_wallet.address,
-            bond_amount=bond_amount,
+            lp_shares=lp_shares,
             market=self,
         )
         self.market_state.apply_delta(market_deltas)
```

Comment:
I noticed that some times we use `self.update_market(market_deltas)` (e.g. in `redeem_withdraw_shares`) and sometimes we use `self.market_state.apply_delta(market_deltas` (e.g. here). I think we should always use `self.update_market()` since it is doing the same thing with additional checks.

Reply:
I don't think I've touched any of that logic in this PR.  I'll do a cleanup PR, I made a [clickup ticket](https://app.clickup.com/t/866a078gp) for this.  Note though that for remove_liquidity we can't use update_market because it can make reserves zero, which update_market protects against.

Reply:
ok sounds good